src/EditorFeatures/CSharpTest2/Recommendations/DelegateKeywordRecommenderTests.cs

```diff
@@ -436,5 +436,45 @@ class C
 {
     delegate*<$$");
         }
+
+        [WorkItem(30784, "https://github.com/dotnet/roslyn/issues/30784")]
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestNotInTypeParameterConstraint_TypeDeclaration1()
+        {
+            await VerifyAbsenceAsync(
+@"class C<T> where T : $$");
+        }
+
+        [WorkItem(30784, "https://github.com/dotnet/roslyn/issues/30784")]
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestNotInTypeParameterConstraint_TypeDeclaration2()
+        {
+            await VerifyAbsenceAsync(
+@"class C<T>
+        where T : $$
```

Comment:
can you have a test of `where T : IList<$$`?

Reply:
Correct

Reply:
@CyrusNajmabadi I got `using Goo = T<$$` test failing while trying to verify this case. I don't understand how `delegate` can be a valid completion for using alias, but not for our case

Reply:
it's likely not valid there either.